storage/mysql/map_storage.go

```diff
@@ -111,6 +111,10 @@ func (m *mySQLMapStorage) begin(ctx context.Context, tree *trillian.Tree) (stora
 		readRevision: -1,
 	}
 
+	if readonly {
+		return mtx, nil
```

Comment:
Maybe add a comment that `readRevision` will be set by a call to `LatestSignedMapRoot`?

Reply:
_might_ be worth making `LatestSignedMapRoot` be a lazy caching getter too?

Reply:
Added the comment; at the moment, lazily getting the map root wouldn't help anything.